immuni_common/helpers/sanic.py

```diff
@@ -226,7 +227,10 @@ async def _wrapper(*args: Any, is_dummy: bool, **kwargs: Any) -> HTTPResponse:
                 await wait_configured_time(
                     mu=config.DUMMY_REQUEST_TIMEOUT_MILLIS, sigma=config.DUMMY_REQUEST_TIMEOUT_SIGMA
                 )
-                return weighted_random(responses)
+                selected_response = weighted_random(responses)
+                if isinstance(selected_response, ApiException):
```

Comment:
are we sure there is no corner cases where we raise an exception not inheriting from ApiException?

Reply:
Currently the only exception that is raised there is the `UnauthorizedOTPException` from the ingestion `/upload` and `/check-otp` endpoints.

The idea is that they have to be exceptions handled by the sanic exception handling system, checks for `ApiExcptions` (the same class that we're using here).

In any other case the system would raise a 500 error, but I don't think there's a use case for simulating 500 errors in dummy calls right? 😄

Reply:
in case of an unexpected exception (e.g. KeyError, whatever), it will be gracefully handled anyway by the Sanic handler, right?

Reply:
Yes, that's a general handler that will run in any case, returning a 500 error.

Reply:
My two cents: the decorator should only handle ApiException since it's a nice way to say "in some cases I'd like a json response", "in some others raise an exception". For any other rare unhandled exception (as of now, we are not aware of any), there's still the Sanic exception handler(s) taking care of it, and as Gab says, it would return 500. All in all, I like the current implementation.

Reply:
yep, I just wanted to be sure we were already covering all the possibilities... LGTM as well 😘